subprojects/gradle/runner/device-provider/src/main/kotlin/com/avito/android/runner/devices/internal/kubernetes/KubernetesReservationClient.kt

```diff
@@ -41,38 +44,42 @@ internal class KubernetesReservationClient(
 
     private var state: State = State.Idling
 
+    private val lock = Mutex()
```

Comment:
Hello, `coroutines` without races. LOL
So we concurrently close channels when other coroutines try to read and write with them
And there is no way except mutex to make this safer